### ISSUE TITLE
Move CRI page into Containers concept section

### ISSUE THREAD

**Issue by sftim** [kubernetes/website#39258]
**This is a Feature Request**

**What would you like to be added**
Move https://kubernetes.io/docs/concepts/architecture/cri/ to be https://kubernetes.io/docs/concepts/containers/cri/

**Why is this needed**
The CRI is not an important detail for people to learn about Kubernetes' overall architecture. Moving it away simplifies the learning journey for people who want to start understanding a high level overview.

**Comments**
/kind cleanup
/language en

If you make this change:
- label the PR as a refactor PR (or remind your reviewer to do that)
- configure a redirect - see file `static/_redirects` - from the old URI to the new one.

**Comment by tengqm** [comment]
I don't agree to this suggestion. The CRI is an important component in the overall architecture but it is pretty transparent to containerized applications and their developers. The current position of `https://kubernetes.io/docs/concepts/containers/` is more for the latter group of people.

**Comment by sftim** [comment]
Can we move that concept page to a different target so that it's not so prominent?

My thinking: we don't introduce CNI or CSI at this point in someone's learning journey. It's OK to _link_ to that as further reading. Putting it in the left nav makes the concept look important to someone who is new to Kubernetes; I think we could imagine a tank of “learning willpower” that our readers are using up as they explore our site.

**Comment by tengqm** [comment]
The current cluster architecture section needs some reorg. Different layers/levels of topics are mixed there with no logic.

<img width="853" alt="image" src="https://user-images.githubusercontent.com/5094756/216867729-739a09a1-5726-440b-84eb-3c777fec79d3.png">
